File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/teleop/TwoJoyOp.java
package org.firstinspires.ftc.teamcode.teleop;

import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

/**
 * Created by josh on 1/7/18.
 */

@SuppressWarnings("unused")
@TeleOp(name = "2 Joy Op", group = "teleop")
public class TwoJoyOp extends RevbotTeleOp {
    @Override
    public void setCurrentDirection(double[] direction) {
        double left = -gamepad1.left_stick_y;
        double right = -gamepad1.right_stick_y;
        double strafe = (gamepad1.left_stick_x + gamepad1.right_stick_x) / 2;

        direction[0] = left;
        direction[1] = right;
        direction[2] = strafe;

        this.currentDirection = direction;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/teleop/TwoJoyOp.kt
package org.firstinspires.ftc.teamcode.teleop

import com.qualcomm.robotcore.eventloop.opmode.TeleOp

/**
 * Created by josh on 1/7/18.
 */

@TeleOp(name = "2 Joy Op", group = "teleop")
class TwoJoyOp : RevbotTeleOp() {
    override fun getUserInput():DoubleArray {
        val direction = DoubleArray(3)
        val left = -gamepad1.left_stick_y.toDouble()
        val right = -gamepad1.right_stick_y.toDouble()
        val strafe = ((gamepad1.left_stick_x + gamepad1.right_stick_x) / 2).toDouble()

        direction[0] = left
        direction[1] = right
        direction[2] = strafe

        return direction
    }
}
